### PR TITLE
Starboardize mutexes in abseil

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/315503544

### CODE REVIEW

**Review by jellefoks — APPROVED** [review]

File: third_party/abseil-cpp/absl/synchronization/mutex_starboard.cc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "absl/synchronization/mutex.h"
+
+#include "starboard/mutex.h"
```
Comment (y4vor) [inline third_party/abseil-cpp/absl/synchronization/mutex_starboard.cc:17]: You can use pthread_mutex_t as long as it is independent from other pthread objects.
We don't have the pthread_cond_t or pthread_once_t yet, but stand alone mutex use should work.

Comment (andrewsavage1) [reply third_party/abseil-cpp/absl/synchronization/mutex_starboard.cc:17]: Resolved offline, it seems the upstream code is doing something much more involved so we're going to go with the starboard APIs until we have broader pthread support

File: third_party/abseil-cpp/absl/synchronization/BUILD.gn [hunk 2]
```diff
@@ -44,52 +44,64 @@ absl_test("kernel_timeout_internal_test") {
 absl_source_set("synchronization") {
   # Synchronization is not Starboardized since at this time it's banned for use
   # in Chromium code.
-  if (!use_cobalt_customizations) {
```
Comment (dahlstrom-g) [inline third_party/abseil-cpp/absl/synchronization/BUILD.gn:47]: We can avoid modifying the lines below (from 48 on).

**Review by dahlstrom-g — APPROVED** [review]
I'm not sure if my approval is needed
for Cobalt 3P Repository Owners—
I did focus on diff-from-upstream formalities.

**Pull request merged**